StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentElement.swift

```diff
@@ -164,8 +164,25 @@ public final class EmbeddedPaymentElement {
     /// Completes the payment or setup.
     /// - Returns: The result of the payment after any presented view controllers are dismissed.
     /// - Note: This method presents authentication screens on the instance's  `presentingViewController` property.
+    /// - Note: This method requires that the last call to `update` succeeded. If the last `update` call failed, this call will fail. If this method is called while a call to `update` is in progress, it waits until the `update` call completes.
     public func confirm() async -> EmbeddedPaymentElementResult {
-        // TODO
+        // Wait for the last update to finish and fail if didn't succeed. A failure means the view is out of sync with the intent and could e.g. not be showing a required mandate.
+        if let currentUpdateTask {
+            switch await currentUpdateTask.value {
+            case .succeeded:
+                break
```

Comment:
if the update succeeds, should we attempt to confirm at this point?

Reply:
Yep, that's what happens in this code (except we haven't written the code to confirm yet).  I'll move the comment in line 185 to make that more obvious.